File: apps/api/src/services/feedback.service.ts
import type { HydratedDocument } from "mongoose";
import type { CreateFeedbackInput, Feedback } from "@believe-ai/shared";
import type { FeedbackDocument } from "@believe-ai/server";
import { feedbackRepository } from "../repositories/feedback.repository.js";

function toDto(doc: HydratedDocument<FeedbackDocument>): Feedback {
  return {
    id: doc._id.toString(),
    userId: doc.userId?.toString() ?? null,
    message: doc.message,
    page: doc.page ?? null,
    status: doc.status as Feedback["status"],
    createdAt: doc.createdAt.toISOString(),
  };
}

export const feedbackService = {
  async submit(userId: string | null, input: CreateFeedbackInput): Promise<Feedback> {
    const doc = await feedbackRepository.create(userId, input.message, input.page ?? null);
    return toDto(doc);
  },
};

File: apps/backend/services/feedback_service.py
"""User-submitted feedback — mirrors apps/api's feedback.service.ts."""

from __future__ import annotations

from bson import ObjectId

from models.feedback import Feedback
from repositories import feedback_repository
from schemas.feedback import FeedbackDto


def _to_dto(doc: Feedback) -> FeedbackDto:
    assert doc.id is not None
    return FeedbackDto(
        id=str(doc.id),
        userId=str(doc.userId) if doc.userId else None,
        message=doc.message,
        page=doc.page,
        status=doc.status,
        createdAt=doc.createdAt.isoformat(),
    )


async def submit(user_id: ObjectId | None, message: str, page: str | None) -> FeedbackDto:
    doc = await feedback_repository.create(user_id, message, page)
    return _to_dto(doc)
